### PR TITLE
rfb: add RFB parser - v5

### PR DESCRIPTION
**Pull request by satta**
Previous PR: #4617

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [#3501](https://redmine.openinfosecfoundation.org/issues/3501)

Describe changes (to previous PR):
- Fix build with unittests enabled.

### CODE REVIEW

File: rust/src/rfb/logger.rs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        pfj.set_integer("depth", tc_server_init.pixel_format.depth as u64);
+        pfj.set_boolean("big_endian", tc_server_init.pixel_format.big_endian_flag != 0);
+        pfj.set_boolean("true_color", tc_server_init.pixel_format.true_colour_flag != 0);
+        pfj.set_integer("red_max", tc_server_init.pixel_format.red_max as u64);
+        pfj.set_integer("green_max", tc_server_init.pixel_format.green_max as u64);
+        pfj.set_integer("blue_max", tc_server_init.pixel_format.blue_max as u64);
+        pfj.set_integer("red_shift", tc_server_init.pixel_format.red_shift as u64);
+        pfj.set_integer("green_shift", tc_server_init.pixel_format.green_shift as u64);
+        pfj.set_integer("blue_shift", tc_server_init.pixel_format.blue_shift as u64);
+        pfj.set_integer("depth", tc_server_init.pixel_format.depth as u64);
```
Comment (jlucovsky) [inline rust/src/rfb/logger.rs]: `depth` is added twice

File: src/detect-rfb-sectype.c [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+ *   [rfb.sectype]:[<|>|<=|>=]<type>;
+ */
+#define PARSE_REGEX "^\\s*(<=|>=|<|>)?\\s*([0-9]+)\\s*$"
+static pcre *parse_regex;
+static pcre_extra *parse_regex_study;
+
+enum DetectRfbSectypeCompareMode {
+    PROCEDURE_EQ = 1, /* equal */
+    PROCEDURE_LT, /* less than */
+    PROCEDURE_LE, /* less than */
```
Comment (jlucovsky) [inline src/detect-rfb-sectype.c]: nit: comment should be "less than or equal" -- same for `PROCEDURE_GE` ... "greater than or equal"

File: src/detect-rfb-sectype.c [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+        Flow *f, uint8_t flags, void *alstate,
+        void *txv, uint64_t tx_id)
+{
+    return DetectEngineInspectGenericList(tv, de_ctx, det_ctx, s, smd,
+                                          f, flags, alstate, txv, tx_id);
+}
+
+static inline int
+SectypeMatch(const uint32_t version,
+        enum DetectRfbSectypeCompareMode mode, uint32_t ref_version)
```
Comment (jlucovsky) [inline src/detect-rfb-sectype.c]: nit: style -- indentation

**Comment by satta** [comment]
New PR: #4622

**Pull request closed without merging**